classes/Utils.php

```diff
@@ -328,21 +328,21 @@ public function update_option( $key = null, $value = false ) {
 	 * @return array|bool|mixed
 	 */
 	public function avalue_dot( $key = null, $array = array(), $default = false ) {
-		$array = (array) $array;
-		if ( ! $key || ! count( $array ) ) {
+		if ( '' === $key || null === $key || ! $this->count( $array ) ) {
```

Comment:
`if ( '' === $key || null === $key || ! $this->count( $array ) )`
Can be written as `if ! empty( $key )`

Reply:
We can't do that because `empty(0)` evaluates to `true`, which means the 0 index would be treated as empty.